department-of-veterans-affairs/va.gov-cms
# VACMS-16233 Reorder VAMC menus

## Description

Relates to #16233

## Acceptance Criteria 
- from #16233
   - [x] Menu items are in the order specified
   - [x] Submenu items follow their parent accordingly
- from #4943
   - [x]  Remove `VA Topeka health care` menu in
   - [x] Remove `VA Leavenworth health care` menu
   
## Screenshots
Prod (old busted) on left, PR (new hotness) on right
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/d02d0589-b8cd-475e-94c6-658f3a1f89b6)

## QA steps

1. Not required but if you want to see the ~520 lines of new code added to codit:menu tools module you can [see it here](https://git.drupalcode.org/project/codit_menu_tools/-/compare/47983ba093816bbdea5ba176011fd1967bb870f1...8d7e391d69580914d7a5180ee12dbef6d0b9c766?from_project_id=127259&straight=false) . 
2. Go to the [tugboat dashboard](https://tugboat.vfs.va.gov/65c66a14067b1809aa0b0586) and click on the php terminal link.  
   - [x] Run the command `drush scr ./scripts/content/VACMS-16233-reorder-vamc-menu-items.php`
   - [x] validate you see something like 
     ![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/82524e5e-9a03-4953-b0f9-ba7f9019c130)
3. Go get lunch... it is going to run for ~ an hour on tugboat.
4. When it is complete:
   - [x] Validate you see a message like 
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/977cee30-2906-44a7-ad54-c9b0550a704f)

5. Go to  System [Washington DC](https://pr17189-mnyjx7or88uud5qagxhpymzarbnbltn2.ci.cms.va.gov/washington-dc-health-care)
   - [x] validate that the primary menu items under "About SYSTEM health care" match the following pattern
   ```
     About us
     Programs
    Research
    Policies
    VA police
    Work with us
    Contact us
   ```
6. Go to  System [Spokane](https://pr17189-mnyjx7or88uud5qagxhpymzarbnbltn2.ci.cms.va.gov/spokane-health-care)
   - [x] validate pattern match ^^ 
7. Go to  System [Greater Los Angeles](https://pr17189-mnyjx7or88uud5qagxhpymzarbnbltn2.ci.cms.va.gov/greater-los-angeles-health-care)
   - [x] validate pattern match ^^ 
8. Go to  System [Orlando](https://pr17189-mnyjx7or88uud5qagxhpymzarbnbltn2.ci.cms.va.gov/orlando-health-care)
   - [x] validate pattern match ^^ 
9. Go to  System [Lovell](https://pr17189-mnyjx7or88uud5qagxhpymzarbnbltn2.ci.cms.va.gov/lovell-federal-health-care)
   - [x] validate pattern match ^^   
10. Login as admin
11. Go to [dblog filtered for codit menu tools](https://pr17189-mnyjx7or88uud5qagxhpymzarbnbltn2.ci.cms.va.gov/admin/reports/dblog?type%5B%5D=codit_menu_tools)
    - [x] validate you only see 6 notices.  They are from misnamed menu items that have already had their names fixed on prod, but tugboat's db does not have the fix yet.
12. Go to [the menu  admin](https://pr17189-mnyjx7or88uud5qagxhpymzarbnbltn2.ci.cms.va.gov/admin/structure/menu)
   - [x] Validate that no menu exists with the word 'topeka' in it (have to check all 4 paginations)
   - [x] Validate that no menu exists with the word 'leavenworth' in it (have to check all 4 paginations)




### Definition of Done
